pkg/object/file.go

```diff
@@ -281,24 +315,19 @@ func (d *filestore) ListAll(prefix, marker string) (<-chan Object, error) {
 			walkRoot = path.Dir(d.root)
 		}
 
-		_ = Walk(walkRoot, func(path string, info os.FileInfo, err error) error {
+		_ = Walk(walkRoot, func(path string, info os.FileInfo, isSymlink bool, err error) error {
 			if runtime.GOOS == "windows" {
 				path = strings.Replace(path, "\\", "/", -1)
 			}
 
 			if err != nil {
-				// skip broken symbolic link
-				if fi, err1 := os.Lstat(path); err1 == nil && fi.Mode()&os.ModeSymlink != 0 {
-					logger.Warnf("skip unreachable symlink: %s (%s)", path, err)
-					return nil
-				}
 				if os.IsNotExist(err) {
 					logger.Warnf("skip not exist file or directory: %s", path)
 					return nil
 				}
 				listed <- nil
 				logger.Errorf("list %s: %s", path, err)
-				return err
+				return nil
```

Comment:
why not return err?

Reply:
if return nil , sync producer will quit loop,  so here need return err to quit recursive walk